index.js

```diff
@@ -84,10 +85,47 @@ Lexer.prototype = {
    * @api private
    */
 
-  tok: function(type, val){
-    return val === undefined ?
-      {type: type, line: this.lineno} :
-      {type: type, line: this.lineno, val: val};
+  tok: function(type, val, columnIncrement){
+    var res = {type: type, line: this.lineno};
+
+    if (this.colno !== null) res.col = this.colno;
```

Comment:
Why would colno ever be `null`?  Unless there's a good reason for it being `null` you should just add it in the same way we do for line.